AnyText/AnyText.Core/Rules/RuleApplication.cs

```diff
@@ -172,7 +172,31 @@ public virtual void Deactivate(ParseContext context)
         /// </summary>
         /// <returns>A collection of parse errors</returns>
         public virtual IEnumerable<ParseError> CreateParseErrors() => Enumerable.Empty<ParseError>();
+        
+        /// <summary>
+        /// Add all CodeLenses of this ruleApplication to a collection 
+        /// </summary>
+        /// <param name="codeLenses">Collection of CodeLenses</param>
+        public virtual void AddCodeLenses(ICollection<CodeLensInfo> codeLenses)
+        {
+            if (Rule.SupportedCodeLenses.Any())
+            {
+                var ruleCodeLenses = Rule.SupportedCodeLenses.Select(a => new CodeLensInfo()
```

Comment:
Eigentlich wäre es auch schön, wenn es zumindest die Möglichkeit zum Filtern gäbe, wenn also nicht immer alle Code Lenses und Code Actions angezeigt werden aber man als Entwickler eine Chance hat, einen Callback anzugeben, der entscheidet, ob die Lens jetzt ausgewertet wird oder nicht.